runtime/interpreter/value_string.go

```diff
@@ -108,6 +100,62 @@ func stringFunctionFromCharacters(invocation Invocation) Value {
 	return NewUnmeteredStringValue(builder.String())
 }
 
+var StringTypeJoinDefaultSeparator = NewUnmeteredStringValue(",")
+
+func stringFunctionJoin(invocation Invocation) Value {
+	argument, ok := invocation.Arguments[0].(*ArrayValue)
+	if !ok {
+		panic(errors.NewUnreachableError())
+	}
+
+	inter := invocation.Interpreter
+
+	switch argument.Count() {
+	case 0:
+		return EmptyString
+	case 1:
+		return argument.Get(inter, invocation.LocationRange, 0)
+	}
+
+	var separator *StringValue
+	if len(invocation.Arguments) > 1 {
+		separator, ok = invocation.Arguments[1].(*StringValue)
+		if !ok {
+			panic(errors.NewUnreachableError())
+		}
+	} else {
+		separator = StringTypeJoinDefaultSeparator
+	}
+
+	// NewStringMemoryUsage already accounts for empty string.
+	common.UseMemory(inter, common.NewStringMemoryUsage(0))
+	var builder strings.Builder
+	first := true
+
+	argument.Iterate(inter, func(element Value) (resume bool) {
+		// Add separator
+		if !first {
+			// -1 to offset the double counting for empty string.
+			common.UseMemory(inter, common.NewStringMemoryUsage(len(separator.Str)))
```

Comment:
comment mentions a `-1`, but the implementation doesn't actually decrement.
Same for line#150

Reply:
Reverted back to manual creation and added a comment. The `-1` seems odd.